.github/workflows/codeql.yml

```
@@ -0,0 +1,64 @@
+name: "CodeQL"
+
+on:
```

I suggest you also run this workflow on `push`, otherwise the results in the Security tab will be outdated until the workflow runs on the schedule. 

```suggestion
on:
  push:
    branches:
      - '*'
```

---

The PR should always be up-to-date before the merge, so it runs twice on the exact same thing.

---

The PR scan results are uploaded for the ref of the PR ([see](https://github.com/traefik/traefik/security/code-scanning?query=pr%3A10003+is%3Aopen) `pr:10003`). The Security tab shows them for the default branch, which will be uploaded on the `push` event mentioned above.

---

The workflow on a PR is not blocking the PR if there are some problems.
It feels weird.

> The Security tab shows them for the default branch

So this is inaccurate for Traefik.

---

What I mean is that, if you do not run the workflow on push, the Security tab won't show the results for the default branch (without having to use the `pr` filter) until it's triggered by the cron. The Security tab doesn't show any results for the default branch yet, because it has not run on `push` yet.